### DISCUSSION TITLE
Negating sensor output

### DISCUSSION THREAD

**Discussion by ghost** [apache/airflow#15228] [Q&A] [answered]
Hello! I have a question. I checked documentation but could not find anything about it.

The question is, how to negate output of a sensor?

For example, I want to use `GCSObjectExistenceSensor` which succeedes when file exists and not if it does not exists. I want it to be the opposite, I want it to succeed when file does not exist. I thought that maybe an option to negate success condition would be in Base Sensor but I cannot find any option for it.

How can I do it?

**Comment by turbaszek** [comment, marked as answer]
There's no such feature but you can simply define your new operator as:
```py
class GCSObjectNotExistenceSensor(GCSObjectExistenceSensor):
  def poke(*args, **kwargs):
    return not super().poke(*args, **kwargs)

```

This seems easy feature to add and possible a useful one - would you like to open a PR?

**Reply by ghost** [reply]
Thank you. You are right, it is easy to create one. I will try opening a PR after I get home.
